homeassistant/components/mqtt/device_tracker/schema_discovery.py

```diff
@@ -75,36 +76,43 @@ class MqttDeviceTracker(MqttEntity, TrackerEntity):
 
     _entity_id_format = device_tracker.ENTITY_ID_FORMAT
 
-    def __init__(self, hass, config, config_entry, discovery_data):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
         """Initialize the tracker."""
-        self._location_name = None
-
+        self._location_name: str | None = None
+        self._value_template: Callable[..., ReceivePayloadType]
```

Comment:
If it's just a type hint, you should be able to set it as a class attribute instead.